Ocean/sediment transport design doc (#793)
MPAS-Dev/MPAS-Model: docs/ocean/design_docs/prognostic-sediment-transport.rst

.. math::

      C(z) w_s + \epsilon_s \frac{dC(z)}{dz} = 0

in which :math:`C(z)` is the suspended sediment concentration (:math:`kg\,m^{-3}`),
:math:`w_s` is the sediment settling velocity (:math:`m\,s^{-1}`),
:math:`\epsilon_s` is the sediment diffusion coefficient (:math:`m^2\,s^{-1}`),
and :math:`z` is the vertical coordinate (:math:`m`) with positive upward.
Assuming the eddy diffusivity varies quadratically with height, the solution of
:math:`C(z)` can be written as

.. math::

      C(z) = C_{ref}\left[ \frac{z_{ref}}{z} \frac{H-z}{H-z_{ref}} \right]^b

in which :math:`C_{ref}` is the near-bottom reference SSC (:math:`kg\,m^{-3}`),
:math:`z_{ref}` is the reference height (:math:`m`) of :math:`C_{ref}`,
:math:`H` is the total water depth (:math:`m`),
and :math:`b` is the suspension parameter called Rouse number, which is calculated by

.. math::

      b = \frac{w_s}{\kappa u_*}

where :math:`\kappa` is the von Karman constant (0.4) and
:math:`u_*` is the shear friction velocity (:math:`m\,s^{-1}`).
The shear stress :math:`\tau = C_d \overline{U}^2={u_*}^2`, so :math:`u_*={C_d}^{0.5} \overline{U}`.

> We provde two options to compute :math:`C_{ref}` and :math:`z_{ref}`:

Reviewer: ```suggestion
We provide two options to compute :math:`C_{ref}` and :math:`z_{ref}`:
```

Reviewer 2: done

Suspended sediment transport also requires the knowledge of the sediment settling velocity.
Sediment settling velocity is a function of the sediment geometry (e.g., particle size, shape and roundness),
the fluid characteristics (density and viscosity), and well as the fluid turbulence velocity.

Two sediment settling veloity formulations are selected to implement into MPAS-Ocean, including

- `Soulsby, 1997 <https://www.icevirtuallibrary.com/doi/abs/10.1680/doms.25844.fm>`_

  .. math::

      w_s = \frac{\nu}{D_{50}}\left[\sqrt{ (10.36^2+1.049D_*^3)}-10.36\right]

  where :math:`\nu` is the kinematic viscosity of water (:math:`m^2\,s^{-1}`).

- `Cheng, 1997 <https://doi.org/10.1061/(ASCE)0733-9429(1997)123:2(149)>`_

  .. math::

      w_s = \frac{\nu}{D_{50}} (\sqrt{25+1.2D_*^2}-5)^{1.5}

  in which :math:`D_* = D_{50}\left[\frac{(s-1)g}{\nu ^2}\right]^{1/3}` is dimensionless grain size.

Implementation
--------------

Implementation: Bedload sediment transport
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/19

Contributors: Zhendong Cao, Xylar Asay-Davis

The sediment bedload transport requires the pre-definitions of the following namelist options:

> - A logical variable to switch ON/OFF the bedload transport: :code:`config_sediment_bedload_enabled`

Reviewer: ```suggestion
- A logical variable to switch ON/OFF the bedload transport: :code:`config_sediment_bedload_enable`
```
For consistency with some of the other config variables

Reviewer 2: I changed all these, thanks. @caozd999 all the flags use `enable` at the end without the d

Author: @mark-petersen Many thanks Mark.

Prognostic non-cohesive sediment transport model
================================================

date: 2021/01/14

Contributors: Zhendong Cao

Summary
-------

The current MPAS-Ocean includes a sediment transport analysis member. But it lacks
the capabilities to simulate the morphological change and the potential interactions
between sediment transport and other system components (e.g., flow field, vegetation dynamics, etc.).
Therefore, we propose a prognostic sediment transport model to enable these missing capabilities.

The proposed model development will focus on the non-cohesive sediment transport.
The model requirements, algorithms and implementation mainly follow the sediment transport
development work in `Warner et al., 2005 <https://doi.org/10.1016/j.cageo.2008.02.012>`_ and
the `COAWST code (USGS version) <https://github.com/jcwarner-usgs/COAWST>`_.
The main model components include

> 1) sediment classes;
> 2) bedload sediment transport;
> 3) suspended sediment transport; and
> 4) morphological change.

Reviewer 3: If you want this to come out formatted like a numbered list, you need blank lines:
```suggestion

1) sediment classes;

2) bedload sediment transport;

3) suspended sediment transport; and

4) morphological change.

```

- `Meyer-Peter-Mueller formulation (1948) <http://resolver.tudelft.nl/uuid:4fda9b61-be28-4703-ab06-43cdc2a21bd7>`_
  This formulation is for the unidirectional flow.

- `Soulsby-Damgaard formulation (2005) <https://doi.org/10.1016/j.coastaleng.2005.04.003>`_ 
  This formulation accounts for the combined effects of currents and waves.

Requirement: Suspended sediment transport
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/17

Contributors: Zhendong Cao

If water flows significantly beyond the threshold of the sediment motion, sediment will be entrained off the bed
and into suspension.
To remain in suspension, the upward turbulence eddies must be greater than the sediment settling velocity.
Suspended mass is exchanged vertically between the water column and the top bed layer.

Suspended sediment transport can be solved either by emprical approaches or by a process-based approach.
The emprical approaches solve SSC profile by the classic Rouse profile assumption.
This approach requires the knowledge of the near-bottom reference SSC,
which can be computed by one of the following two methods:

> - `Zyserman and Fredsoe, 1994 <https://ascelibrary.org/doi/pdf/10.1061/%28ASCE%290733-9429%281994%29120%3A9%281021%29>`_
> - `Goldstein et al., 2014 <https://esurf.copernicus.org/articles/2/67/2014/esurf-2-67-2014-discussion.html>`_

Reviewer 3: ```suggestion
- `Zyserman and Fredsoe, 1994 <https://ascelibrary.org/doi/pdf/10.1061/%28ASCE%290733-9429%281994%29120%3A9%281021%29>`_

- `Goldstein et al., 2014 <https://esurf.copernicus.org/articles/2/67/2014/esurf-2-67-2014-discussion.html>`_
```

The process-based approach solves the suspended sediment transport by the three-dimensional advection-diffusion equation
(i.e., the scaler transport equation),  with an additional source/sink term for sediment vertical settling and exchange
with the bed.

Suspended sediment transport requires the knowledge of the sediment settling velocity.
Sediment settling velocity is a function of the sediment geometry (e.g., particle size, shape and roundness),
the fluid characteristics (density and viscosity), and well as the fluid turbulence velocity.
Two sediment settling veloity formulations are selected to implement into MPAS-Ocean, including

> - `Soulsby, 1997 <https://www.icevirtuallibrary.com/doi/abs/10.1680/doms.25844.fm>`_
> - `Cheng, 1997 <https://doi.org/10.1061/(ASCE)0733-9429(1997)123:2(149)>`_

Reviewer 3: ```suggestion
- `Soulsby, 1997 <https://www.icevirtuallibrary.com/doi/abs/10.1680/doms.25844.fm>`_

- `Cheng, 1997 <https://doi.org/10.1061/(ASCE)0733-9429(1997)123:2(149)>`_
```

Requirement: Morphological change
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/20

Contributors: Zhendong Cao

Morphological change here stands for the change of sea floor elevation (bathymetry) due to the convergence/divergence of net
sediment flux during a time period. The Exner equation (`Paola and Voller, 2005 <https://doi.org/10.1029/2004JF000274>`_)
is implemented to compute the morphological change.

Algorithm Design
---------------------------

Algorithm Design: Sediment classes
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/16

Contributors: Zhendong Cao

The number of sediment classes will be defined manually and considered as a dimension of sediment-related variables.

Algorithm Design: Bedload sediment transport
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/16

Contributors: Zhendong Cao

In practice, a bedload coefficient should be defined to adjust the bedlad transport rate (see `COAWST code here
<https://github.com/jcwarner-usgs/COAWST/blob/master/ROMS/Nonlinear/Sediment/sed_bedload.F#L608-L612>`_).

The Meyer-Peter-Mueller and Soulsby-Dammgaard share the same expression of the bedload transport rate

> .. math::
>     q_b = \Phi \sqrt{(s-1)g D_{50}^3}\rho_s

Reviewer 3: ```suggestion
.. math::

    q_b = \Phi \sqrt{(s-1)g D_{50}^3}\rho_s
```

Reviewer 3: Same in many other places.

Author: I will work through the document to fix them.

in which :math:`q_b` is the bedload transport rate (:math:`kg m^{-1}s^{-1}` ), :math:`\Phi` is dimensionless bedload
transport rate, :math:`s=\rho_s-\rho_w` is the sediment specific density in water, :math:`\rho_s` and :math:`\rho_w`
are the sediment denisty and the water density, respectively (:math:`kg m^{-3}`), :math:`g` is the gravitational
acceleration (:math:`m s^{-2}`), :math:`D_{50}` is the median grain diameter (:math:`m`).

The main difference of the two formulations is the computation of :math:`\Phi`, which is a function of another
dimensionless parameter called Shields parameter :math:`\theta`

> .. math::
> 	\theta = \frac{\tau}{g(s-1)D_{50}}

Reviewer 3: ```suggestion
.. math::

	\theta = \frac{\tau}{g(s-1)D_{50}}
```

Reviewer 3: You're also using tabs for indenting in a few places.  The document should not have any tabs, just spaces.

Author: Thanks. Is there any specific number of spaces I should place for the indenting format?

Reviewer 3: I follow the python specification of 4 spaces but I don't think RST/sphinx cares.

Reviewer 3: I looked into this further.  The preference is for 3 spaces: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/lists.html#unordered-list-inside-ordered-list

bedload transport flux (:math:`m^2s^{-1}`) in :math:`x` and :math:`y` direction, respectively; :math:`C_{source,b}` is the
bottom suspended transport rate (:math:`kg m^{-2}s^{-1}`).

Implementation
--------------

Implementation: Sediment classes
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/14

Contributors: Zhendong Cao

The number of sediment classes `nSedimentClasses` will be defined as a dimension before the
definitions of the sediment properties. The definition will be in a `Registry.xml` file and the code is

.. code::

	dim name='nSedimentClasses' unit='unitless'
	    description='the number of sediment classes considered in the model'

Implementation: Bedload sediment transport
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/19

Contributors: Zhendong Cao

The sediment bedload transport requires the pre-definitions of the following namelist options:

- A logical variable to switch ON/OFF the bedload transport: :code:`config_sediment_bedload`

- A character variable to define the bedload transport method: :code:`config_sediment_bedload_formulation`

>   And there are three options for it:
> 	* :code:`Engelund-Hansen`
> 	* :code:`Meyer-Peter-Mueller`
> 	* :code:`Soulsby-Damgaard`

Reviewer 3: ```suggestion
  And there are three options for it:

        * :code:`Engelund-Hansen`

	* :code:`Meyer-Peter-Mueller`

	* :code:`Soulsby-Damgaard`
```

These can be defined in **Registry.xml** as follows:

.. code::

  <nml_record name="sediment_transport" mode="init;forward">
        <nml_option name="config_sediment_bedload" type="logical" default_value=".false." units="unitless"
                    description="Controls if sediment bedload transport is computed."
                    possible_values=".true. or .false."
        />
        <nml_option name="config_sediment_bedload_formulation" type="chracter" default_value="Meyer-Peter-Mueller" units="unitless"
                    description="Select the sediment bedload transport formulation"
                    possible_values=" 'Engelund-Hansen', 'Meyer-Peter-Mueller', 'Soulsby-Damgaard'"
	/>
  </nml_record>

Bedload transport also requires the definitions of the following variables:

> 
> - Pre-defined sediment-related variables
> 	* sediment grain diameter: :code:`sedimentGrainDiameter(nSedimentClasses, nCells)`
> 	* sediment grain density: :code:`sedimentGrainDensity(nSedimentClasses, nCells)`
> 	* sediment class fraction on bed layer: :code:`sedimentClassFraction(nSedimentClasses)`
> 	* bedload transport coefficient: :code:`sedimentBedloadCoefficient(nSedimentClasses)`
> - Other variables
> 	* bedload transport rate: :code:`sedimentBedloadTransportRate(nCells)`

Reviewer 3: ```suggestion

- Pre-defined sediment-related variables

  * sediment grain diameter: :code:`sedimentGrainDiameter(nSedimentClasses, nCells)`

  * sediment grain density: :code:`sedimentGrainDensity(nSedimentClasses, nCells)`

  * sediment class fraction on bed layer: :code:`sedimentClassFraction(nSedimentClasses)`

  * bedload transport coefficient: :code:`sedimentBedloadCoefficient(nSedimentClasses)`

- Other variables

  * bedload transport rate: :code:`sedimentBedloadTransportRate(nCells)`
```

Reviewer 3: I guess you get the idea.  Many of these lists need this kind of spacing.

Contributors: Zhendong Cao

Summary
-------

The current MPAS-Ocean includes a sediment transport analysis member. But it lacks
the capabilities to simulate the morphological change and the potential interactions
between sediment transport and other system components (e.g., flow field, vegetation dynamics, etc.).
Therefore, we propose a prognostic sediment transport model to enable these missing capabilities.

The proposed model development will focus on the non-cohesive sediment transport.
The model requirements, algorithms and implementation mainly follow the sediment transport
development work in `Warner et al., 2005 <https://doi.org/10.1016/j.cageo.2008.02.012>`_ and
the `COAWST code (USGS version) <https://github.com/jcwarner-usgs/COAWST>`_.
The main model components include

1) sediment classes;

2) bedload sediment transport;

3) suspended sediment transport;

4) morphological change.

Sediment classes determine the properties of sediment-related variables.
Bedload and suspended sediment transport are the two main transport forms considered in this development.
Morphological change provides the trigger for interactions between different model components.

> The primary implimentation challenges lie in the suspended sediment transport component.

Author: corrected.

> It requires to solve the three-dimensional scalar equation for suspended sediment with

Author: corrected.

its own source/sink terms, which need to resolve the deposition and resuspension at each
time step.

> The sucessful implementation will enable MPAS-Ocean to simulate the sediment transport (especially

Author: corrected.

> the computation of suspended sediment cocentration (SSC)) with given initial and boundary conditions,

Author: corrected.

Sediment settling velocity is a function of the sediment geometry (e.g., particle size, shape and roundness),
the fluid characteristics (density and viscosity), and well as the fluid turbulence velocity.
Two sediment settling veloity formulations are selected to implement into MPAS-Ocean, including

- `Soulsby, 1997 <https://www.icevirtuallibrary.com/doi/abs/10.1680/doms.25844.fm>`_

- `Cheng, 1997 <https://doi.org/10.1061/(ASCE)0733-9429(1997)123:2(149)>`_

Requirement: Morphological change
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/20

Contributors: Zhendong Cao

Morphological change here stands for the change of sea floor elevation (bathymetry) due to the convergence/divergence of net
sediment flux during a time period. The Exner equation (`Paola and Voller, 2005 <https://doi.org/10.1029/2004JF000274>`_)
is implemented to compute the morphological change.

Algorithm Design
---------------------------

Algorithm Design: Sediment classes
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/16

Contributors: Zhendong Cao

The number of sediment classes will be defined manually and considered as a dimension of sediment-related variables.

> Algorithm Design: Bedload sediment transport

Author: I plan to implement both and let user choose.

.. math::

      C(z) w_s + \epsilon_s \frac{dC(z)}{dz} = 0

in which :math:`C(z)` is the suspended sediment concentration (:math:`kg m^{-3}`), :math:`w_s` is the sediment settling velocity
(:math:`ms^{-1}`), :math:`\epsilon_s` is the sediment diffusion coefficient (:math:`m^2s^{-1}`), and :math:`z` is the vertical
coordinate (:math:`m`) with positive upward. Assuming the eddy diffusivity varying parabolically with height, the solution of
:math:`C(z)` can be written as

.. math::

      C(z) = C_{ref}\left[ \frac{z_{ref}}{z} \frac{H-z}{H-z_{ref}} \right]^b

in which :math:`C_{ref}` is the near-bottom reference SSC (:math:`kg m^{-3}`),
:math:`z_{ref}` is the reference height (:math:`m`) of :math:`C_{ref}`,
:math:`H` is the total water depth (:math:`m`),
and :math:`b` is the suspension parameter called Rouse number, which is calculated by

.. math::

      b = \frac{w_s}{\kappa u_*}

where :math:`\kappa` is the von Karman constant (0.4) and :math:`u_*` is the shear friction velocity (:math:`ms^{-1}`).
The shear stress :math:`\tau = C_d \overline{U}^2={u_*}^2`, so :math:`u_*={C_d}^{0.5} \overline{U}`.

We provde two options to compute :math:`C_{ref}` and :math:`z_{ref}`:

> - *Zyserman and Fredsoe, 1994*
> 
> .. math::
> 
>       C_{ref} = \frac{0.331(\theta - 0.045)^{1.75}}{1+0.72(\theta-0.045)^{1.75}}
> 
> and :math:`z_{ref}=2D_{50}`. Again, :math:`\theta` is the Shields parameter and
> :math:`D_{50}` is the sediment median grain diameter.
> 
> - *Goldstein et al., 2014*
> 
> .. math::
> 
>       C_{ref} = \left[ \frac{0.328U_b}{0.0688+1000D_{50}}\right]^2
> 
> at height :math:`z_{ref}=0.01 m`, where :math:`U_b` is the magnitude of the bottom velocity (:math:`ms^{-1}`)
> 

Reviewer 3: Based on this link: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/lists.html#code-images-and-other-content-inside-lists

```suggestion
- *Zyserman and Fredsoe, 1994*

  .. math::

      C_{ref} = \frac{0.331(\theta - 0.045)^{1.75}}{1+0.72(\theta-0.045)^{1.75}}

  and :math:`z_{ref}=2D_{50}`. Again, :math:`\theta` is the Shields parameter and
:math:`D_{50}` is the sediment median grain diameter.

- *Goldstein et al., 2014*

  .. math::

      C_{ref} = \left[ \frac{0.328U_b}{0.0688+1000D_{50}}\right]^2

  at height :math:`z_{ref}=0.01 m`, where :math:`U_b` is the magnitude of the bottom velocity (:math:`ms^{-1}`)

```

Reviewer 3: That is, everything but the "bullet" lines need to be indented 2 spaces.

Author: Got it.

**Process-based approach**

The proces-based approach solves SSC by the three-dimensional advection-diffusion equations, with an additional source/sink term
for vertical settling and exchange with the bed

.. math::

      \frac{\partial{H(z)C(z)} }{\partial t} + \nabla \cdot \left( H(z)C(z) \bf{u}\right) = \nabla \cdot \left ( \epsilon_s \nabla H(z)C(z) \right) + C_{source}

with the addtional term written as

.. math::

      C_{source} = -w_s C(z) + E_s

in which :math:`H(z)` is the layer thickness (:math:`m`), :math:`t` is time (:math:`s`), :math:`\nabla` represents
the advection process, :math:`\bf{u}` is the velocity vector (:math:`ms^{-1}`),
:math:`E_s` is the erosion source (:math:`kg m^{-2} s^{-1}`), which is computed as

.. math::

      E_s = E_0(1-\phi)\frac{\tau_{sf}-\tau_{ce}}{\tau_{ce}}

where :math:`E_0` is the bed erodibility constant (:math:`kg m^{-2}s^{-1}`), :math:`\phi` is the porosity of the bed layer,
:math:`\tau_{ce}` is the bed critical erosion stress (:math:`m^2s^{2}`)

The methods to compute sediment settling velocity include

- *Soulsby 1997*

  .. math::

      w_s = \frac{\nu}{D_{50}}\left[\sqrt{ (10.36^2+1.049D_*^3)}-10.36\right]

> where :math:`\nu` is the kinematic viscosity of water (:math:`m^2s^{-1}`)

Reviewer 3: This needs to be indented, too:
```suggestion
  where :math:`\nu` is the kinematic viscosity of water (:math:`m^2s^{-1}`)
```

- *Cheng 1997*

  .. math::

      w_s = \frac{\nu}{D_{50}} (\sqrt{25+1.2D_*^2}-5)^{1.5}

> in which :math:`D_* = D_{50}\left[\frac{(s-1)g}{\nu ^2}\right]^{1/3}` is dimensionless grain size.

Reviewer 3: ```suggestion
  in which :math:`D_* = D_{50}\left[\frac{(s-1)g}{\nu ^2}\right]^{1/3}` is dimensionless grain size.
```

Algorithm Design: Morphological change
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/20

Contributors: Zhendong Cao

Exner equation reads as

.. math::

      (1-\phi)\frac{\partial Z_b}{\partial t} + \left(\frac{\partial q_{bx}}{\partial x}+\frac{\partial q_{by}}{\partial y}\right) = - \frac{C_{source,b}}{\rho_s}

in which :math:`\phi` is bed porosity, :math:`Z_b` is the bathymetry (:math:`m`), :math:`q_{bx}` and :math:`q_{by}` are bottom
bedload transport flux (:math:`m^2s^{-1}`) in :math:`x` and :math:`y` direction, respectively; :math:`C_{source,b}` is the
bottom suspended transport rate (:math:`kg m^{-2}s^{-1}`).

Implementation
--------------

Implementation: Sediment classes
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/14

Contributors: Zhendong Cao

The number of sediment classes `nSedimentClasses` will be defined as a dimension before the
definitions of the sediment properties. The definition will be in a `Registry.xml` file and the code is

.. code::

	dim name='nSedimentClasses' unit='unitless'
	    description='the number of sediment classes considered in the model'

Implementation: Bedload sediment transport

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Author: I think these are intermediate variables that maybe don't need to be defined externally.

.. code::

  <nml_record name="sediment_transport" mode="init;forward">
        <nml_option name="config_sediment_bedload" type="logical" default_value=".false." units="unitless"
                    description="Controls if sediment bedload transport is computed."
                    possible_values=".true. or .false."
        />
        <nml_option name="config_sediment_bedload_formulation" type="chracter" default_value="Meyer-Peter-Mueller" units="unitless"
                    description="Select the sediment bedload transport formulation"
                    possible_values=" 'Engelund-Hansen', 'Meyer-Peter-Mueller', 'Soulsby-Damgaard'"
	/>
  </nml_record>

Bedload transport also requires the definitions of the following variables:

- Pre-defined sediment-related variables
	
  * sediment grain diameter: :code:`sedimentGrainDiameter(nSedimentClasses, nCells)`

  * sediment grain density: :code:`sedimentGrainDensity(nSedimentClasses, nCells)`

  * sediment class fraction on bed layer: :code:`sedimentClassFraction(nSedimentClasses)`

  * bedload transport coefficient: :code:`sedimentBedloadCoefficient(nSedimentClasses)`

- Other variables
	

>   * bedload transport rate: :code:`sedimentBedloadTransportRate(nCells)`

Author: `\rho_w` should be from the lowest ocean layer. I will remove the wave effects for just now.

	
  * :code:`None`

  * :code:`Zyserman-Fredsoe`

  * :code:`Goldstein`

  If this is set as :code:`None`, then the suspended transport is solved by the process-based approach.

- A character-type variable to define the method to compute the sediment settling velocity: :code:`config_sediment_settling_formulation`. There are three options for it:

  * :code:`None`

  * :code:`Soulsby`

  * :code:`Cheng`

  If this is set as :code:`None`, then the sediment settling velocity is pre-defined manually.

These can be defined in **Registry.xml** as follows:

.. code::

  <nml_record name="sediment_transport" mode="init;forward">
        <nml_option name="config_sediment_suspended" type="logical" default_value=".false." units="unitless"
                    description="Controls if sediment bedload transport is computed."
                    possible_values=".true. or .false."
        />
        <nml_option name="config_sediment_suspended_formulation" type="chracter" default_value="None" units="unitless"
                    description="Select the sediment bedload transport formulation"
                    possible_values=" 'None', 'Zyserman-Fredsoe', 'Goldstein'"

> 	/>

Author: Yes, that's right. I will add `config_sediment_settling_formulation`in the registry file.

  </nml_record>

Suspended sediment transport also requires the definitions of the following variables:

- Pre-defined sediment-related variables

  * sediment settling velocity: :code:`sedimentSettlingVelocity(nSedimentClasses)`

  * sediment critical erosion shear: :code:`sedimentCriticalErosionShear(nSedimentClasses)`

  * sediment erodibility constant: :code:`sedimentErodibilityConstant(nSedimentClasses)`

  * sediment bed porosity: :code:`sedimentBedPorosity(nSedimentClasses)`

- Other variables

>   * sediment suspended concentration: :code:`sedimentSuspendedConcentration(nSedimentClasses, nCell)`

Reviewer 3: Again, a corresponding variable name from Algorithm Design: Suspended sediment transport would be helpful for each variable here.

The bottom velocity might warrant some discussion.  For one, the velocity magnitude isn't necessarily available at cell centers (but it can be computed from the kinetic energy).  For another, you may want to average over more than one layer to smooth out jumps depending on what vertical coordinates and bathymetries this capability supports (see my suggested requirement related to this.)

> Implementation: Morphological change

Reviewer 3: This may not seem like it but it might be the hardest change to make.  You may want to meet us with me, @dengwirda, @cbegeman and @stephenprice to discuss this because it may relate to work we are doing for ice-shelf cavities.  Currently, `bottomDepth` is the (positive-down) bathymetry in MPAS-Ocean, but it isn't allowed to change in time.  It is also intimately tied to the z* vertical coordinate that is currently the default and is implicitly assumed in at least a few places in the code that @cbegeman has discussed with us (related to the computation of vertical velocities between layers).

A change to the model to allow `bottomDepth` to change with time could be a pretty big project and could potentially be a bit disruptive.  @matthewhoffman may also be interested in this work, since it also relates to his early career project.  In sort, this might be a big enough project to need a design document all on its own with input from a lot of different people.

Reviewer: @caozd999, @lconlon and I have had a conversation about implementing morphological change to the bathymetry. Correct me if I'm wrong, but I think our conclusion was that this development wasn't strictly necessary for ICoM's goals and that it likely wouldn't be feasible in the [current development timeframe](https://icom.atlassian.net/wiki/spaces/ICOM/pages/101711907/Overview). I'd suggest removing it from this design doc and if we find ourselves ahead of the timeline or if this development seems necessary for ICoM then we create a new design document for morphological change alone.

Reviewer 4: I'm trying to remember that conversation now....would erosion and deposition still work correctly if we don't change the bed elevation in the model? I remember that we decided that changing coastline morphology was out of the scope of this project, but I can't remember where we landed on bed elevation.

Author: @cbegeman you are right, we don't have to implement the morphological change component for ICoM now; I put it in this doc for a more general model component introduction.  As @xylar and @cbegeman suggested, I will remove morphological change from the design doc, and we can prepare a single design doc for when necessary. But it is good to know that this would influence/interest other people if we tend to do it; seems to me a necessary model development direction.

Author: @lconlon we would just assume the erosion and deposition are not strong enough to significantly change either the bed layer sediment mass or the sediment layer thickness. So it would work. And yes you are right - we had a discussion before and the coastline morphology won't be in the scope either.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Date last modified: 2021/01/20

Contributors: Zhendong Cao

> A logical variable need to be defined to swith ON/OFF the morphological change: :code:`config_morphological_change`.

Reviewer 3: ```suggestion
A logical variable need to be defined to switch ON/OFF the morphological change: :code:`config_morphological_change_enabled`.
```

Author: corrected.
